x-pack/plugins/infra/public/containers/logs/log_flyout.tsx

```diff
@@ -21,25 +21,26 @@ export enum FlyoutVisibility {
 
 export interface FlyoutOptionsUrlState {
   flyoutId?: string | null;
+  flyoutIndex?: string | null;
   flyoutVisibility?: string | null;
   surroundingLogsId?: string | null;
 }
 
 export const useLogFlyout = () => {
   const { sourceId } = useLogSourceContext();
   const [flyoutVisible, setFlyoutVisibility] = useState<boolean>(false);
-  const [flyoutId, setFlyoutId] = useState<string | null>(null);
+  const [flyoutRef, setFlyoutRef] = useState<{ id: string; index: string } | null>(null);
```

Comment:
What do you think about changing the `*Ref` suffix? That way we can reserve it to React refs.

Maybe something more specific, like `flyoutLogItem`, or `flyoutLogLine`